### PR TITLE
Open browser to show Help in settings

### LINKED ISSUE #526: Settings: Implement Help
## Description
Same as the old app.

[iOS Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19800511) [Android Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19932960) [Notion page.](https://www.notion.so/toggl/Settings)

## Dependencies
`Settings: Add settings view` #512
## Definition of done
- [ ] Implement the logic
- [ ] implement the UI

### PR DESCRIPTION
**Pull request by simonrozsival**
### Summary
This PR adds the ability to open a browser at a specific URL.

### Technical considerations
I used the `backStack` state property. I'm not sure this is the best solution since we can't catch the action in some middleware (could hack it somehow with a "higher order reducer", but I'm not sure that's a good solution).

I used the `MainActivity` to open the browser when the top of the back stack points to a Browser URL. I had to handle returning back to the app to pop the browser route from the stack.

I had to deal with the back button issues, because otherwise the app behaved weirdly.

### Why is it done like this and not some other way?
My knowledge of all possibilities of our architecture and Android APIs is still limited, so I'm really not sure if this is a good solution. I'm looking forward to your reviews and I suppose I'll rewrite the PR based on the feedback.

### Relationships
Closes #526 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🚫 not yet

### CODE REVIEW

File: settings/src/main/java/com/toggl/settings/domain/SettingsReducer.kt [hunk 1]
```diff
@@ -114,4 +118,9 @@ class SettingsReducer @Inject constructor(
                 it.copy(sendFeedbackRequest = loadable)
             }
         }
+
+    private fun MutableValue<SettingsState>.navigateTo(route: Route) =
+        mutateWithoutEffects<SettingsState, SettingsAction> {
+            copy(backStack = backStack.push(route))
+        }
```
Comment (heytherewill) [inline settings/src/main/java/com/toggl/settings/domain/SettingsReducer.kt:125]: Nice extension 👌

**Comment by simonrozsival** [comment]
@heytherewill I updated the PR and removed the `BackStackReducer` and the related code. I'm still not sure this is a good way to implement this feature, but at least it's quite simple. I'm not sure where to put the URL right now and it seemed overcomplicated. I'll revisit it tomorrow.

**Comment by semanticer** [comment]
I'm taking over this issue. I don't think we need a separate fragment just to launch a browser. I would simply `startActivity(openURL)` from SettingsFragment and I wouldn't even define a special route for it. `navigateTo` extension looks dope tho, I would keep that for sure.

**Comment by semanticer** [comment]
Closing this since there is a new PR implementing this https://github.com/toggl/android/issues/526

**Pull request closed without merging**